src/resolve-config.js

```diff
@@ -1,26 +1,54 @@
 "use strict";
 
 const cosmiconfig = require("cosmiconfig");
+const editorconfig = require("editorconfig");
 const minimatch = require("minimatch");
 
 const asyncWithCache = cosmiconfig("prettier");
 const asyncNoCache = cosmiconfig("prettier", { cache: false });
 const syncWithCache = cosmiconfig("prettier", { sync: true });
 const syncNoCache = cosmiconfig("prettier", { cache: false, sync: true });
 
+function editorConfigToPrettier(filePath) {
+  const editorConfig = editorconfig.parseSync(filePath);
+  const result = {};
+  result.useTabs = editorConfig.indent_style === "tab";
+  result.tabWidth =
+    editorConfig.indent_size || editorConfig.tab_width || result.tabWidth;
+  result.printWidth = editorConfig.max_line_length || result.printWidth;
```

Comment:
There's no need for `|| result.tabWidth` and `|| result.printWidth`, is it? They are always goind to be `undefined`.

Couldn't we just do:

```js
const editorConfig = editorconfig.parseSync(filePath);
return {
  useTabs: ...,
  tabWidth: ...,
  printWidth: ...
}
```

While being an odd configuration, isn't it valid to set `indent_size`, `tab_width` and `max_line_length` to `0`, and as such we should respect that?

Reply:
> There's no need for || result.tabWidth and || result.printWidth, is it? They are always goind to be undefined.

Yeah, that was also causing a bug. Not sure why I did it that way, but it's fixed in 54a5dcb1945f90f86dfff647e8183a450c806a12

> While being an odd configuration, isn't it valid to set indent_size, tab_width and max_line_length to 0, and as such we should respect that?

Good point, I'll try to fix that.